packages/hoppscotch-data/src/collection/v/2.ts

```diff
@@ -53,6 +52,15 @@ export default defineVersion({
         authType: "inherit",
       },
       headers: [],
+      folders: old.folders.map((folder) => {
+        const result = HoppCollection.safeParseUpToVersion(folder, 2)
+
+        if (result.type !== "ok") {
+          throw new Error("Failed to migrate child collections")
+        }
```

Comment:
This doesn't necessarily require additional safeguards since any schema mismatch about child collections falling in an invalid version range would be caught right at the parsing phase and won't reach this block.